--- /tmp/sunxi-common.h	2013-12-29 14:06:42.199693905 +0100
+++ include/configs/sunxi-common.h	2013-12-29 14:07:11.395693156 +0100
@@ -74,7 +74,11 @@
 /* A10 has 1 banks of DRAM, we use only bank 1 in U-Boot */
 #define CONFIG_NR_DRAM_BANKS		1
 #define PHYS_SDRAM_0			CONFIG_SYS_SDRAM_BASE
+#ifdef CONFIG_SUN7I
+#define PHYS_SDRAM_0_SIZE		0x80000000
+#else
 #define PHYS_SDRAM_0_SIZE		0x40000000
+#endif
 #if 0
 /* Nand config */
 #define CONFIG_NAND
